### PR TITLE
DB schema changes

### PR DESCRIPTION
**Pull request by tsnowlan**
First pass. Still testing, but wanted to get some input.

Main things:
* took table primary key generation out of the code and put it back into the DB
* still get Users.count during user creation, but that's just to allow setting `is_admin` on first user
* added indexes
* `report_meta.report_meta_value` extended to  1024, as I know I was running into import issues with that one.

Related to the last change: postgres, mysql, and sqlite all support text fields without a specified size. SQLAlchemy has a `Unicode` column type that takes care of the unicode handling we were talking about with the py3 fixes and has negligible overhead compared to varchar(X) ([on postgres at least](https://www.postgresql.org/docs/current/static/datatype-character.html)). Thoughts on converting all the `varchar`s and `String` columns to `Unicode`? It's a breaking change, so either nuke / re-import or a patching script (alter table/column for postgres, export / sed / re-import for sqlite) would be needed for existing installs.

### CODE REVIEW

File: megaqc/api/utils.py [hunk 1]
```diff
@@ -183,9 +175,9 @@ def handle_report_data(user, report_data):
             existing_plot_config = db.session.query(PlotConfig).filter(PlotConfig.config_type==report_data['report_plot_data'][plot]['plot_type'], PlotConfig.config_name==plot, PlotConfig.config_dataset==dataset_name).first()
             if not existing_plot_config:
                 new_plotcfg_cnt += 1
-                config_id = PlotConfig.get_next_id()
+#                 config_id = PlotConfig.get_next_id()
```
Comment (ewels) [inline megaqc/api/utils.py]: Can we just delete these commented lines?

File: megaqc/model/models.py [hunk 2]
```diff
@@ -50,8 +50,8 @@ class ReportMeta(db.Model, CRUDMixin):
     __tablename__ =  'report_meta'
     report_meta_id = Column(Integer, primary_key=True)
     report_meta_key = Column(String(80), nullable=False)
-    report_meta_value = Column(String(80), nullable=False)
-    report_id = Column(Integer, ForeignKey('report.report_id'))
+    report_meta_value = Column(String(1024), nullable=False)
+    report_id = Column(Integer, ForeignKey('report.report_id'), index=True)
 
     @staticmethod
     def get_next_id():
```
Comment (ewels) [inline megaqc/model/models.py]: Do we need these methods any more?

File: megaqc/user/models.py [hunk 3]
```diff
@@ -65,10 +65,6 @@ def __init__(self, password=None, **kwargs):
         else:
             self.password = None
 
-
-        if self.user_id == 1:
```
Comment (ewels) [inline megaqc/user/models.py]: Better 👍

**Comment by ewels** [comment]
Fine by me to update to `Unicode` if we're not losing compatibility (which it sounds like we're not)..

File: megaqc/api/utils.py (hunk 1 above)
Comment (tsnowlan) [reply megaqc/api/utils.py]: Yeah, that was just a lazy way to see where `X_id` was being used so I could remove it. I'll be doing some cleanup after I finish testing.

Comment (ewels) [reply megaqc/api/utils.py]: Yeah, I figured that was the case as you did say that it was a first pass.. 👍

**Comment by tsnowlan** [comment]
Okay, should be all clean and ready except for the column type conversions. Do you want a conversion script for afterwards, or are you good with just reloading data? I'll putting one together for our postgres instance anyway.

**Comment by tsnowlan** [comment]
So, this might be a SQLalchemy thing, but existing tables have no problems being read/written by the new field definitions. Very neat.

The last two commits have some patch files:
* `scripts/varchar2text.sql`: updates the table defs to use text instead of varchars
* `scripts/resync_pg_seqs.sh`: existing postgres installs will still have all their sequences set at 1. running this will match them to table values.

I think this is ready for a final review.

**Comment by ewels** [comment]
Many thanks for this, looks great! 👍

**Pull request merged**